### DISCUSSION TITLE
Osteosarcoma annotation (SCPCP000023)

### DISCUSSION THREAD

**Discussion by patelgrp** [AlexsLemonade/OpenScPCA-analysis#587] [Propose a new analysis]
### Proposed analysis

We will annotate the collection of osteosarcoma datasets covered in SCPCP000023 (n=54) samples. Our pipeline involves a series of cleanup, QC filtering, and automated annotation steps prior to expert-guided refinement. Specifically:
- we will start with count matrices generated by ALSF
- we will perform doublet prediction using demuxafy
- we will filter low-quality cells/nuclei (using number of genes/cell and percent of UMIs from mitochondrial genes as filter thresholds)
- we will perform copy number calling using inferCNV and, if aligned bam files are available, numbat
- we will then use a 3 tiered approach to annotate non-malignant clusters:
1. using a pure merge of all the datasets, we will identify cells/nuclei that high grade of mixing as measured using the LISI index. These clusters have a high likelihood of being non-malignant stroma because inter-individual variation in those cell types, in our experience, is low.
2. we perform automated annotation using SingleR with the human cell atlas as a reference. Those cells with annotation to endothelium or immune cells will be annotated as non-malignant
3. finally, those clusters with absence of inferred copy number alterations will be annotated as non-malignant

Finally, we will use Seurat's layer integration pipeline to generate an integrated dataset (in our hands scVI or reciprocal PCA perform the best).

### Scientific goals

To annotate the non-malignant and malignant cells within SCPCP000023 

### Methods or approach

- we will start with count matrices generated by ALSF
- we will perform doublet prediction using demuxafy
- we will filter low-quality cells/nuclei (using number of genes/cell and percent of UMIs from mitochondrial genes as filter thresholds)
- we will perform copy number calling using inferCNV and, if aligned bam files are available, numbat
- we will then use a 3 tiered approach to annotate non-malignant clusters:
1. using a pure merge of all the datasets, we will identify cells/nuclei that high grade of mixing as measured using the LISI index. These clusters have a high likelihood of being non-malignant stroma because inter-individual variation in those cell types, in our experience, is low.
2. we perform automated annotation using SingleR with the human cell atlas as a reference. Those cells with annotation to endothelium or immune cells will be annotated as non-malignant
3. finally, those clusters with absence of inferred copy number alterations will be annotated as non-malignant

Finally, we will use Seurat's layer integration pipeline to generate an integrated dataset (in our hands scVI or reciprocal PCA perform the best).

### Existing modules

Yes, this module is based on an existing module of Ewing sarcoma samples in https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/292#discussion-6427891.

### Input data

The analysis will use count matrices extracted from the SingleCellExperiment objects for SCPCP000023

### Scientific literature

https://www.nature.com/articles/s41467-020-20059-6
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10515803/
https://www.frontiersin.org/journals/oncology/articles/10.3389/fonc.2022.732862/full

**Comment by jashapiro** [comment]
This sounds great! I'll just note here that as we have already done some QC filtering, you may be able to skip some of the initial steps within this analysis (and the others you have proposed) if you use those results. In general, we would like to keep uniform filtering and QC across modules where we can. 

We do not yet have doublet filtering as part of our workflow, however, so it would be great to have those results available for other modules. Perhaps we could break that out into a separate module to run across all samples? Note that we have started to play with other demultiplexing methods in https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/doublet-detection (see also the discussion in https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/364), but we have not yet finalized those analyses. If `demuxify` seems like a better solution, or just one we want to make available as well, we could add it into that module, and then you could use those results as part of your analysis here. It may be that breaking out the doublet analysis to a separate module is a later process as well; it may be partially a discussion of what is most convenient.

Finally, I'll note that because we do not have aligned bam files as part of the data generally available for these projects, we would have to figure out how best to integrate such results if you find that `numbat` is something you want to incorporate.

**Reply by patelgrp** [reply]
Good to know about the QC filtering - in that case, we'll probably just generate a few plots with metrics and avoid more filtering. 

For doublet filtering, we've moved our pipelines to [demuxafy](https://demultiplexing-doublet-detecting-docs.readthedocs.io/en/latest/). It's basically a wrapper for most of the doublet methods out there. I like it because it's easy to run 3 or 4 different doublet methods and then use the consensus, since there is surprising variation between doublet-calling methods. Once the data is in hand, I'll be able to take a look and decide if it makes more sense to keep the doublets and just annotate them as 'doublets', or whether it makes sense to remove them from downstream analysis and annotation. One added quirk is that osteoclasts in bone are multinucleated, so I'd hate to remove them from analysis by mistakenly calling them as an artifact.

Regarding `numbat`, I've become a fan of the joint allele and copy-number calling in that package. The allele calling has a low sensitivity, but it's very specific for identifying cells with somatic variants (i.e. transformed malignant cells). I think it's kinda designed to work with Cellranger output, and I think trying to port your alevin-aligned data might be a challenge. For now, we can run through `inferCNV` (which will take forever) or we might try `copykat` (which has worked poorly for low mutation burden tumors in my hands, but osteosarcoma might be the ideal use-case).

Thanks, and I'm excited to working with y'all again!

**Comment by Jen-OMalley** [comment]
Hi @patelgrp. Just chiming in to say thank you for sharing your analysis ideas! The rest of the team will be reviewing your proposed analyses, and we're looking forward to discussing further. We will set up an AWS account for you shortly. Once we do, you should receive an email with an invitation to finish setting up your account. I'll reach out again when you should be expecting to see this. We'll be back in touch with next steps within 3 business days.

In the meantime, let us know if you have any questions. Looking forward to working together!

**Reply by Jen-OMalley** [reply]
@patelgrp your AWS account has been created and you should receive an email to complete setup. [Here are instructions for setting up AWS](https://openscpca.readthedocs.io/en/latest/software-platforms/aws/joining-aws/)!

**Comment by allyhawkins** [comment]
Thanks for putting this together @patelgrp! I had a few follow up questions, but generally this approach and outline sounds good to me. 

It sounds like you might actually have a pipeline of sorts in place for this? Is this something that already exists somewhere and you plan to apply it to this dataset and the other two datasets you filed discussion posts for (#588 and #589)? 

For the CNV calling, how do you plan on identifying the normal cells to use as the reference in `InferCNV`? Or do you plan on using it where you do not provide any reference cells? 

You also mention using the LISI index on clusters, which I really like! I just wanted to note that although we do provide cluster assignments in the processed `SingleCellExperiment` objects on the Portal, I would recommend re-doing your own clustering. We used default parameters to apply the same clustering approach to all samples on the Portal, so I don't expect those to be the most reliable clusters. 

You also mention running this pipeline before doing expert-guided refinement. Do you know what that would entail at this point? I assume this is the part that will be more disease-specific and involve more manual inspection and validation of the cell-type assignments. 

I would also encourage you to check out these helpful sections of our documentation to get started: 

[Technical setup](https://openscpca.readthedocs.io/en/latest/technical-setup/)
[Contributing to Analysis](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/)

Please let me know if you have specific questions on how to get started!

**Comment by allyhawkins** [comment]
Hi @patelgrp, I just wanted to follow up to help you get started. 

I know Josh started to mention this in an earlier comment, but in an effort to keep the analysis across modules uniform and transparent, we ask that you start your analysis with the processed objects available on the ScPCA Portal (`_processed.rds` or `_processed_rna.h5ad`). These objects have already undergone removal of empty droplets, filtering of low quality cells, and normalization. See the [documentation on the ScPCA Portal](https://scpca.readthedocs.io/en/stable/processing_information.html) for more information about how these objects were processed. This ensures that all analyses in `OpenScPCA` have the same starting point. 

When you are ready to get started, you can download the data using the [`download-data.py` script in the repository](https://openscpca.readthedocs.io/en/latest/getting-started/accessing-resources/getting-access-to-data/), which will download the processed objects by default.

As part of `OpenScPCA` we have also annotated doublets in all samples using [`scDblFinder`](https://bioconductor.org/packages/release/bioc/html/scDblFinder.html). Currently, we are in the process of running `scDblFinder` on all samples, and the results will be made available to you via S3. These results will include a TSV file with the output from `scDblFinder`. If you plan to annotate and remove doublets, please use these results. If you would like to use these results before they are available, you may also run the available script in the [`doublet-detection` module](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/analyses/doublet-detection/scripts/01a_run-scdblfinder.R).

If you need to use filtering, normalization, or doublet detection methods that are different from the methods that are already defined within the project, please provide your rationale and evidence that supports the alternative method.

 To get started, we recommend that you pick one of the three projects you proposed and start an analysis module using the data from that project. Once the code has been developed and added to the repository, it should be easy to apply it to all the projects you plan to work on. 

 When you are ready to start your analysis, please follow the below steps to [start contributing to the project](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/):

- [Follow the technical setup instructions found in the `OpenScPCA` documentation](https://openscpca.readthedocs.io/en/latest/technical-setup/).
- [File an issue to track the initiation of your analysis module](https://openscpca.readthedocs.io/en/latest/communications-tools/github-issues/).
- [Initiate your module](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/creating-a-module/) and [file a pull request](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/creating-pull-requests/).

After you have initiated your module, you will be ready to continue with the rest of the analysis that you proposed. I would recommend that you break up your work into the following steps to start, where each bullet point would be an issue and at least one subsequent pull request: 

- Run copy number inference to identify tumor cells
- Perform clustering in samples 
- Calculate the LISI index on all clusters
- Annotate non-malignant cells with `SingleR`

For more information on contributing to the project, I recommend you review these sections of the documentation: 

- [More on analysis modules](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/)
- [Working with Git](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/working-with-git/)
- [Scoping and creating pull requests](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/creating-pull-requests/)
